scale/node/resources/resource.py

```diff
@@ -0,0 +1,97 @@
+"""Defines the node resource classes"""
+from __future__ import unicode_literals
+
+from abc import ABCMeta
+
+
+class Resource(object):
+    """Abstract class for a node resource
+    """
+
+    __metaclass__ = ABCMeta
+
+    def __init__(self, name, resource_type):
+        """Constructor
+
+        :param name: The name of the resource
+        :type name: string
+        :param resource_type: The type of the resource
+        :type resource_type: string
+        """
+
+        self.name = name
+        self.resource_type = resource_type
+
+    def copy(self):
```

Comment:
I've started using `@abstractmethod` decorator instead of raise... I believe that is the proper way to do this now and future proof.